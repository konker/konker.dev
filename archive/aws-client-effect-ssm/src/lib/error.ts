export const TAG = 'SsmError';

export type SsmError = {
  readonly _tag: typeof TAG;
  readonly _Params: unknown;
  readonly message: string;
  readonly cause: unknown;
};

export const toSsmError =
  <I>(params: I) =>
  (x: unknown): SsmError => {
    return {
      _tag: TAG,
      _Params: params,
      message: typeof x === 'object' && x && 'message' in x ? (x as any).message : String(x),
      cause: x,
    };
  };
