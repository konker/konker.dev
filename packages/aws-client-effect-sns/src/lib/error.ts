export const TAG = 'SnsError';

export type SnsError = {
  readonly _tag: typeof TAG;
  readonly _Params: unknown;
  readonly message: string;
  readonly cause: unknown;
};

export const toSnsError =
  <I>(params: I) =>
  (x: unknown): SnsError => {
    return {
      _tag: TAG,
      _Params: params,
      message: typeof x === 'object' && x && 'message' in x ? (x as any).message : String(x),
      cause: x,
    };
  };
