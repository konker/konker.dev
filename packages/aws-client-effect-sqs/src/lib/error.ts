export const TAG = 'SqsError';

export type SqsError = {
  readonly _tag: typeof TAG;
  readonly _Params: unknown;
  readonly message: string;
  readonly cause: unknown;
};

export const toSqsError =
  <I>(params: I) =>
  (x: unknown): SqsError => {
    return {
      _tag: TAG,
      _Params: params,
      message: typeof x === 'object' && x && 'message' in x ? (x as any).message : String(x),
      cause: x,
    };
  };
