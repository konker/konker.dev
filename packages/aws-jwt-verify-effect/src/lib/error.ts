export const TAG = 'JwtVerifyError';

export type JwtVerifyError = {
  readonly _tag: typeof TAG;
  readonly message: string;
  readonly cause: unknown;
};

export const toJwtVerifyError = (x: unknown): JwtVerifyError => {
  return {
    _tag: TAG,
    message: typeof x === 'object' && x && 'message' in x ? (x as any).message : String(x),
    cause: x,
  };
};
