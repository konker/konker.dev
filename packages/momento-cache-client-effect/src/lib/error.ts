import type { TinyError } from '@konker.dev/tiny-error-fp';
import { toTinyError } from '@konker.dev/tiny-error-fp';

export const ERROR_TAG = 'MomentoClientError' as const;
export type ERROR_TAG = typeof ERROR_TAG;

export type MomentoClientError = TinyError<ERROR_TAG>;
export const toMomentoClientError = toTinyError<ERROR_TAG>(ERROR_TAG);
