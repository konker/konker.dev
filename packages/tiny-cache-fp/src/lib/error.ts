import type { TinyError } from '@konker.dev/tiny-error-fp';
import { toTinyError } from '@konker.dev/tiny-error-fp';

export const ERROR_TAG = 'CacheError' as const;
export type ERROR_TAG = typeof ERROR_TAG;

export type CacheError = TinyError<ERROR_TAG>;
export const toCacheError = toTinyError<ERROR_TAG>(ERROR_TAG);
