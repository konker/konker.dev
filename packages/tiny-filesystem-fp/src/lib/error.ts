export const TAG = 'TinyFileSystemError';

export type TinyFileSystemError = {
  readonly _tag: typeof TAG;
  readonly message: string;
  readonly cause: unknown;
};

export function toTinyFileSystemError(x: unknown): TinyFileSystemError {
  return {
    _tag: TAG,
    message: typeof x === 'object' && x && 'message' in x ? (x as any).message : String(x),
    cause: x,
  };
}
