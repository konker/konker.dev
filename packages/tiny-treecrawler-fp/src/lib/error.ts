import { toTinyError } from '@konker.dev/tiny-error-fp';

export const ERROR_TAG = 'TinyTreeCrawlerError';
export type ERROR_TAG = typeof ERROR_TAG;

export const toTinyTreeCrawlerError = toTinyError<ERROR_TAG>(ERROR_TAG);
export type TinyTreeCrawlerError = ReturnType<typeof toTinyTreeCrawlerError>;
