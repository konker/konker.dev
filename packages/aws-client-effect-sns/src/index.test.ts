import type { PublishCommandInput } from '@aws-sdk/client-sns';
import * as snsClient from '@aws-sdk/client-sns';
import { SNSClient } from '@aws-sdk/client-sns';
import { mockClient } from 'aws-sdk-client-mock';
import { pipe } from 'effect';
import * as Effect from 'effect/Effect';
import { beforeAll, beforeEach, describe, expect, it } from 'vitest';

import * as unit from './index.js';
import { SNSClientDeps } from './index.js';
import { TAG } from './lib/error.js';

const snsMock = mockClient(snsClient.SNSClient);

describe('aws-client-effect-sns', () => {
  let deps: unit.SNSClientDeps;

  beforeAll(() => {
    deps = unit.SNSClientDeps.of({
      snsClient: new snsClient.SNSClient({}),
    });
  });

  describe('Error tag', () => {
    it('should export as expected', () => {
      expect(unit.SNS_ERROR_TAG).toEqual('SnsError');
    });
  });

  // ------------------------------------------------------------------------
  describe('defaultSNSClientFactory', () => {
    it('should work as expected', async () => {
      expect(unit.defaultSNSClientFactory({})).toBeInstanceOf(SNSClient);
    });
  });

  // ------------------------------------------------------------------------
  describe('defaultSNSClientFactoryDeps', () => {
    it('should work as expected', async () => {
      const actualEffect = pipe(
        unit.SNSClientFactoryDeps,
        Effect.map((deps) => deps.snsClientFactory),
        unit.defaultSNSClientFactoryDeps
      );
      const actual = Effect.runSync(actualEffect);
      expect(actual).toBeInstanceOf(Function);
    });
  });

  // ------------------------------------------------------------------------
  describe('defaultSNSClientDeps', () => {
    it('should work as expected', async () => {
      const actualEffect = pipe(
        unit.SNSClientDeps,
        Effect.map((deps) => deps.snsClient),
        unit.defaultSNSClientDeps({})
      );
      const actual = Effect.runSync(actualEffect);
      expect(actual).toBeInstanceOf(SNSClient);
    });
  });

  // ------------------------------------------------------------------------
  describe('PublishCommand', () => {
    beforeEach(() => {
      snsMock.reset();
    });

    it('should work as expected', async () => {
      snsMock.on(snsClient.PublishCommand).resolves({ MessageId: 'test-message-id' });

      const params: PublishCommandInput = { TopicArn: 'test-topic-arn', Message: 'test-message' };
      const expected = { MessageId: 'test-message-id', _Params: params };
      const command = pipe(unit.PublishCommandEffect(params), Effect.provideService(SNSClientDeps, deps));
      await expect(Effect.runPromise(command)).resolves.toStrictEqual(expected);
      expect(snsMock.calls().length).toEqual(1);
    });

    it('should fail as expected', async () => {
      snsMock.on(snsClient.PublishCommand).rejects({ $metadata: { httpStatusCode: 404 } });

      const params: PublishCommandInput = { TopicArn: 'test-topic-arn', Message: 'test-message' };
      const command = pipe(unit.PublishCommandEffect(params), Effect.provideService(SNSClientDeps, deps));
      await expect(Effect.runPromise(command)).rejects.toThrowErrorMatchingSnapshot(JSON.stringify({ _tag: TAG }));
      expect(snsMock.calls().length).toEqual(1);
    });
  });
});
