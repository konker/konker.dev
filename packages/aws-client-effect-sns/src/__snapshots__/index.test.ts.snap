// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`aws-client-effect-sns > PublishCommand > should fail as expected > {"_tag":"SnsError"} 1`] = `[(FiberFailure) Error: {"_tag":"SnsError","_Params":{"TopicArn":"test-topic-arn","Message":"test-message"},"message":"","cause":{"$metadata":{"httpStatusCode":404}}}]`;
